### PR TITLE
Fixed Many regex in the Validation method miss a trailing newline

### LINKED ISSUE #9992: Many regex in the Validation method miss a trailing newline
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.3.11
* Platform and Target: macOS X Sierra + PHP 5.6.28

### What you did
Validation method using regular expression like `email()` verified string with a newline at the end.

```php
<?php

namespace App\Test\Model\Validation;

use Cake\TestSuite\TestCase;
use Cake\Validation\Validation;

class ValidationTest extends TestCase
{
    public function testEmail()
    {
        // This assertion will failed.
        self::assertFalse(Validation::email("user@example.com\n"));
    }
}
```

### What happened
return true.

### What you expected to happen
return false.

Similar problem fixed: #1177

### PR DESCRIPTION
**Pull request by tenkoma**
fixed #9992

### CODE REVIEW

**Comment by dereuromark** [comment]
Not sure that is the right solution.
I usually always trim my post data coming in before patching it.
This way all the validation methods kicking in then work on the expected data.

The validation rules you try to fix indeed expect the data to be clean from leading and trailing whitespace, but so do all others, including "notEmpty/notBlank" etc to respond properly.

So I think the current behavior is somewhat correct in to have these expectations, and it is the task of the developer to find a way to sanitize them beforehand. Otherwise we are over-complicating those validation methods, and will still not be able to catch all kind of "whitespace" noise issues.

File: tests/TestCase/Validation/ValidationTest.php [hunk 1]
```diff
@@ -1102,6 +1103,7 @@ public function testDateMmddyyyy()
         $this->assertFalse(Validation::date('11.31.2006', ['mdy']));
         $this->assertFalse(Validation::date('11/31/2006', ['mdy']));
         $this->assertFalse(Validation::date('11 31 2006', ['mdy']));
+        $this->assertFalse(Validation::date("12-27-2006\n", ['mdy']));
```
Comment (markstory) [inline tests/TestCase/Validation/ValidationTest.php]: My concern is that these are no longer dates. They are dates + a new line. If provided to a database or external system this _could_ fail as there is additional whitespace.

Comment (tenkoma) [inline tests/TestCase/Validation/ValidationTest.php]: That's right.
I understood that it is a task that should be handled separately from `Validation` class validation.

**Comment by tenkoma** [comment]
@dereuromark 
> I usually always trim my post data coming in before patching it.
> This way all the validation methods kicking in then work on the expected data.

> and it is the task of the developer to find a way to sanitize them beforehand.

I think that it is the developer's task to remove unnecessary data before validation. me too.
I understand.
Thank you for reviewing. Close this pull request.

**Comment by dereuromark** [comment]
I think we are missing some best practice documentation on the subject.
It would be great if you can help us here.

**Pull request closed without merging**